src/Sqids.php

```diff
@@ -772,40 +769,36 @@ public function decode(string $id): array
 
     protected function shuffle(string $alphabet): string
     {
-        $chars = str_split($alphabet);
-
-        for ($i = 0, $j = count($chars) - 1; $j > 0; $i++, $j--) {
-            $r = ($i * $j + ord($chars[$i]) + ord($chars[$j])) % count($chars);
-            [$chars[$i], $chars[$r]] = [$chars[$r], $chars[$i]];
+        for ($i = 0, $j = strlen($alphabet) - 1; $j > 0; $i++, $j--) {
+            $r = ($i * $j + ord($alphabet[$i]) + ord($alphabet[$j])) % strlen($alphabet);
+            [$alphabet[$i], $alphabet[$r]] = [$alphabet[$r], $alphabet[$i]];
```

Comment:
Manipulation of individual characters of the string, instead of using an array of chars.